pkg/agent/client/client.go

```diff
@@ -171,7 +171,14 @@ func (c *client) RenewSVID(ctx context.Context, csr []byte) (*X509SVID, error) {
 	})
 	if err != nil {
 		c.release(connection)
-		c.c.Log.WithError(err).Error("Failed to renew agent")
+		log := c.c.Log
+		if s, ok := status.FromError(err); ok {
```

Comment:
can you add a helper function to reduce code? (and use constants), something like:
```
func addErrorFields(log logrus.FieldLogger, err error) logrus.FieldLogger {
   if s, ok := status.FromError(err); ok {
        log = log.WithFields(logrus.Fields{                
		telemetry.StatusCode:    s.Code(),
		telemetry.StatusMessage: s.Message(),
	})
   }
   return log
}
```
and you can call
```
   log := addErrorFields(c.c.Log, err)
```

Reply:
Yes, that's what I asked Taylor in a previous comment but I'm not sure where to create that function. Is there a helper package or something similar that is suitable to hold this kind of stuff?

Reply:
add as private on the same file, since, or do you think you need to use in on different packages?

Reply:
For now it will only be used on the same package.

Reply:
Fixed on 6c8d983, let me know in case there's more suggestions :)